clang/lib/CIR/Dialect/IR/CIRDialect.cpp

```diff
@@ -1338,6 +1338,22 @@ LogicalResult cir::VTableAddrPointOp::verify() {
 /// the name of the attribute in ODS.
 static StringRef getLinkageAttrNameString() { return "linkage"; }
 
+static void printExtraFunctionAttr(mlir::cir::ExtraFuncAttributesAttr attr,
+                                   OpAsmPrinter &os) {
+  if (attr.getElements().empty())
+    return;
+  os << " extra_attributes {";
```

Comment:
Instead of `extra_attributes {...}` I'd prefer if we are a bit more terse, like `extra({...})`? I believe we can leverage on existing `DictionaryAttr` printer to automatically print the dict attribute, all we need in the tablegen file would be (assuming you also added "extra" to the defining `CIR_Attr` name):
```
let assemblyFormat = [{ `(` $elements `)` }];
```

Reply:
extra({...}) sounds good.

I'm not sure we can leverage assemblyFormat. I tried it. As mentioned above, it always print the mnemonic even for an empty attribute.